### DISCUSSION TITLE
Question about the differences about "max throughput" mode and "low latency" mode.

### DISCUSSION THREAD

**Discussion by Wizmann** [microsoft/msquic#4658] [Q&A] [answered]
What are the design considerations for `QUIC_EXECUTION_PROFILE_TYPE_MAX_THROUGHPUT` compared to `QUIC_EXECUTION_PROFILE_LOW_LATENCY`?

From the code, I observed that in "max throughput" mode, a separate thread is created for each worker, and "soft-affinity" is set to an ideal processor. 

Could you clarify under what circumstances the thread owned by the worker is executed? It seems that the `CXPLAT_WORKER` directly invokes the callbacks of `QUIC_WORKER`, and the worker thread only wakes up on events such as new connections or operations. 

Additionally, how does this approach enhance throughput in "max throughput" mode compared to the "low latency" option?

**Comment by nibanks** [comment, marked as answer]
Max throughput, as you point out, creates an additional thread just for the QUIC layer to execute on. This allows for parallel execution of the QUIC (encryption) and UDP layers, which, for low connection count scenarios, can see up to a 2x throughput improvement. But if you have LOTS of connections, you will likely only see contention, resulting in worse performance. This is why the default is low latency. So, unless you know you have a low connection count (CPU count > connection count), I recommend low latency mode.
